Realm/RLMSyncPermissions_Private.hpp

```diff
@@ -0,0 +1,41 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2017 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMSyncPermissions.h"
+
+#import "sync/sync_permission.hpp"
+
+@interface RLMSyncPermissionValue ()
+
+- (instancetype)initWithPermission:(realm::Permission)permission;
+
+- (realm::Permission)rawPermission;
+
+@end
+
+// `RLMSyncPermissionValue` is subclassed so that future types of
+// permission values can be implemented in the future.
+@interface RLMSyncUserIDPermissionValue : RLMSyncPermissionValue
```

Comment:
Do we really need to subclass `RLMSyncPermissionValue`? I don't see what that gains us.